src/coreservices.h

```diff
@@ -37,8 +37,12 @@ class CoreServices : public QObject {
     ~CoreServices() = default;
 
     void initializeSettings();
-    // FIXME: should be private, but WMainMenuBar needs it initialized early
     void initializeKeyboard();
+    void initializeLogging();
+
+    // The short first run that is done without start up screen
+    void preInitialize(QApplication* pApp);
+    // The secondary long run which should be called after displaying the start up screen
     void initialize(QApplication* pApp);
```

Comment:
Can we just store the pApp instead of having to pass it twice?

Reply:
Storing would violate the C++ guidline to not store a plain pointer.  That's why I like to keep it like that.

Reply:
Do even need it? It can be accessed in a signleton manner, via `qApp`, although I generally dislike such an approach because it makes it unclear what kinds of data the method depends on.

Reply:
I have decided against it, because using the singleton App obfuscates the control file even more.